File: mini-bank/backend/src/minibank/schemas/summary.py
"""Account finance summary — KPI + by-category breakdown."""
from decimal import Decimal

from pydantic import BaseModel


class CategoryTotal(BaseModel):
    category: str
    total: Decimal


class AccountSummary(BaseModel):
    month: str
    inflow: Decimal
    outflow: Decimal
    mtd_balance: Decimal
    by_category: list[CategoryTotal]

File: mini-bank/frontend/src/api/summary.ts
import { api } from "./client";

export interface CategoryTotal {
  category: string;
  total: string;
}

export interface AccountSummary {
  month: string;
  inflow: string;
  outflow: string;
  mtd_balance: string;
  by_category: CategoryTotal[];
}

export const summaryApi = {
  getMonth: (accountId: string, month?: string) =>
    api
      .get<AccountSummary>(`/accounts/${accountId}/summary`, {
        params: month ? { month } : {},
      })
      .then((r) => r.data),
};
